Gems/Atom/Feature/Common/Code/Include/Atom/Feature/Utils/FrameCaptureBus.h

```diff
@@ -18,14 +19,15 @@ namespace AZ
 {
     namespace Render
     {
-        AZ_ENUM_CLASS(FrameCaptureResult,
-            None,
-            Success,
-            FileWriteError,
-            InvalidArgument,
-            UnsupportedFormat,
-            InternalError
-        )
+        //! The errors met in initializing the frame capture.
+        //! It is used for script Ebus calls to provide a richer debug environment.
+        struct FrameCaptureError
+        {
+            AZ_TYPE_INFO(FrameCaptureError, "{9459AC1D-B0EE-4D89-9EEC-6A65790C76BF}");
+            static void Reflect(ReflectContext* context);
+
+            AZStd::string m_errorMessage;
+        };
```

Comment:
does this need to be a whole class? or can it just be a string?

Reply:
It can be a string for now. But the idea of doing so is to avoid API changes again in the future.
For example, if we want to add extra error data, or change it to an enum to save string operation, we add those without affecting anyone using it, or we can write much easier deprecation code within the class.